### ISSUE TITLE
Flexibility

### ISSUE THREAD

**Issue by hellowin** [Josebaseba/sails-multiple-assets-example#1]
Actually I already use this approach, but it have no flexibility. For example, for one of my projects, it has 2 different assets that needs to separate, one is app page, one is admin page. Another project have 3 different assets, app, admin, and special authentication page.

With this approach I must register each assets linker/ tasks with editing grunt configuration. I think it's not pretty flexible. I have expectation that Sails linker can matching our pattern. Maybe, at folder with `assets` prefix. So it will become `<!-- SCRIPTS [MODULE] --><!-- SCRIPTS [MODULE] END -->`. So we will have flexibility to add or remove assets as we want.

Or maybe to not makes Sails directory dirty, it can be wrapped to single folder like.

```
sails
  |- assets
  |- assets-module
    |- admin
    |- app
    |- ...
```

What do you think?

**Comment by Josebaseba** [comment]
That's a good idea make the stuff more automatic, let me think about it. 

Btw, have you seen that I've added a functionality that adds the package version in the minified .js/.css file? I'm doing to force the user to download the latest version of the file, avoiding the problems with the cache. So if I made changes in the .js I'm sure that changing the package.json version the user is going to download it.

What do think about that?

**Comment by hellowin** [comment]
Something like this `dest: '.tmp/public/min/app.' + version + '.production.min.js'` right? That's good idea.

Back to topic about flexibility, is my proposed feature is executable? My thought that it needs to have folder name conventions so some grunt task can scan it's pattern like

```
sails
  |- assets
  |- assets-module
    |- admin
    |- app
```

it will scanned into `admin` and `app` modules and automatically create a linker to something like `<!--SCRIPTS APP--><!--SCRIPTS APP END-->`. But to make it convenience, before modifying Sails core repository, it's better to create a Sails Hooks. So people just needs to do something like `npm install sails-hook-modular-assets` and that feature will be available without breaking somethings. But is Sails hooks capable to modify Grunt tasks? Or do you have any other approach? The `versioning system` that you created also available to implemented in that feature/ module (I hope hooks is capable to do that).

**Comment by Josebaseba** [comment]
A hook couldn't access to grunt logic. Maybe it has to be a template generator that overwrites the default tasks/ folder for a tasks folder that down what we want. 

Look at this repo: https://github.com/balderdashy/sails-generate-frontend

And to this one too: https://github.com/balderdashy/sails-generate-generator/

``` javascript
  npm install sails-generate-multiple-assets
  sails generate multiple-assets
```

Example: https://github.com/kasperisager/sails-generate-auth

And the pattern I will use for the assets path would be something like that: 

```
sails
  |- assets
    |- admin
       |- js
       |- css
       |- img
       ...
    |- app
       |- js
       |- css
       |- img
       ...
```

And the tags would be ,  etc...

**Comment by hellowin** [comment]
Alright, today I've learned little bit about sails's hooks and yes it can't access Grunt logic, also it can't access controller's logic (am I right?).

So, it would be nice if `sails-generate-multiple-assets` is exist right? Maybe I'll work on it later in my leisure time and reopen my closed issue after I finished with my generator. Or maybe you would help me creating that kind of thing? Btw, thank you anyway to teach me about modifying Grunt's logic in this repo, it's very helpful.
